cmd/_integration-tests/transport/http_test.go

```diff
@@ -400,6 +400,67 @@ func TestStrangeRPCName(t *testing.T) {
 	}
 }
 
+func TestHTTPErrorStatusCodeAndNilHeaders(t *testing.T) {
+	// See handlers/handlers.go for implementation
+	// Returns status code http.StatusTeapot
+	req, err := http.NewRequest("GET", httpAddr+"/status/code", strings.NewReader(""))
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "cannot construct http request"))
+	}
+
+	client := &http.Client{}
+	httpResp, err := client.Do(req)
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "cannot make end http request"))
+	}
+	defer httpResp.Body.Close()
+
+	got, want := httpResp.StatusCode, http.StatusTeapot
+	if got != want {
+		t.Fatalf("Expected status code:`%d`, Got status code: `%d`", want, got)
+	}
+}
+
+func TestHTTPErrorStatusCodeAndHeaders(t *testing.T) {
+	// See handlers/handlers.go for implementation
+	// Returns status code http.StatusOk and headers
+	// Foo: Bar
+	// Test: A, B
+
+	req, err := http.NewRequest("GET", httpAddr+"/status/code/and/headers", strings.NewReader(""))
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "cannot construct http request"))
+	}
+
+	client := &http.Client{}
+	httpResp, err := client.Do(req)
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "cannot make end http request"))
+	}
+	defer httpResp.Body.Close()
+
+	got, want := httpResp.Header, map[string][]string{
+		"Foo":  []string{"Bar"},
+		"Test": []string{"A", "B"},
+	}
+
+	for k := range want {
+		if len(got[k]) == 0 {
+			t.Fatalf("Expected Header `%s:%s`; Got not header with key: `%s`", k, want[k], k)
```

Comment:
Typo: "Got not"

This is a controlled environment, so this seems okay, but testing that the length is zero is not indicative of a lack of map membership--for this to be certain, you must use the two-return-value map access, `v, ok := m[k]`.

Curious about the backtick quotes in the error strings. NBD, but seems kind of arbitrary since backtick quotes are not typically used in general.